src/OneSignal.ts

```diff
@@ -587,10 +587,8 @@ export default class OneSignal {
       Log.warn("User is not subscribed, cannot remove external user id.");
       return;
     }
-    await Promise.all([
-      OneSignal.database.setExternalUserId(undefined),
-      OneSignal.context.updateManager.sendExternalUserIdUpdate(undefined),
-    ]);
+    await OneSignal.context.updateManager.sendExternalUserIdUpdate(undefined),
```

Comment:
Why did we remove `Promise.all` here?

Reply:
Talked about this, the order matters. This is because we need to ensure that `external_id_auth_hash` is in the local db as it will be read and sent in the REST API call. Once it is done then we can remove it with the next line from the local db.